src/QsCompiler/CompilationManager/EditorSupport/CodeActions.cs

```diff
@@ -182,57 +219,84 @@ private static IEnumerable<TextEdit> OpenDirectiveSuggestions(this FileContentMa
+                .SelectMany(d => file.TypeNamespaceSuggestions(d.Range.Start.ToQSharp(), compilation, out _));
+            return file
+                .OpenDirectiveSuggestions(lineNr, idSuggestions.Concat(typeSuggestions))
+                .Select(edit => (edit.NewText.Trim().Trim(';'), file.GetWorkspaceEdit(edit)));
+        }
+
+        /// <summary>
+        /// Returns workspace edits for correcting the casing or capitalization of unknown identifiers.
+        /// </summary>
+        private static IEnumerable<(string, WorkspaceEdit)> UnknownIdCaseSuggestions(
+            this FileContentManager file, CompilationUnit compilation, IReadOnlyCollection<Diagnostic> diagnostics)
+        {
+            (string, WorkspaceEdit) SuggestedIdEdit(string suggestedId, Lsp.Range range)
             {
-                return Enumerable.Empty<(string, WorkspaceEdit)>();
+                var edit = new TextEdit { Range = range, NewText = suggestedId };
+                return ($"Replace with \"{suggestedId}\".", file.GetWorkspaceEdit(edit));
             }
 
-            var suggestionsForIds = unknownCallables.Select(d => d.Range.Start)
-                .SelectMany(pos => file.NamespaceSuggestionsForIdAtPosition(pos.ToQSharp(), compilation, out var _));
-            var suggestionsForTypes = unknownTypes.Select(d => d.Range.Start)
-                .SelectMany(pos => file.NamespaceSuggestionsForTypeAtPosition(pos.ToQSharp(), compilation, out var _));
-            return file.OpenDirectiveSuggestions(lineNr, suggestionsForIds.Concat(suggestionsForTypes).ToArray())
-                .Select(edit => (edit.NewText.Trim().Trim(';'), file.GetWorkspaceEdit(edit)));
+            var suggestionsForIds =
+                from diagnostic in diagnostics
+                where DiagnosticTools.ErrorType(ErrorCode.UnknownIdentifier)(diagnostic)
+                from id in file.IdCaseSuggestions(diagnostic.Range.Start.ToQSharp(), compilation)
+                select SuggestedIdEdit(id, diagnostic.Range);
+            var suggestionsForTypes =
+                from diagnostic in diagnostics
+                where DiagnosticTools.ErrorType(ErrorCode.UnknownType)(diagnostic)
+                from type in file.TypeCaseSuggestions(diagnostic.Range.Start.ToQSharp(), compilation)
+                select SuggestedIdEdit(type, diagnostic.Range);
+            return suggestionsForIds.Concat(suggestionsForTypes);
         }
 
+        /// <summary>
+        /// Returns a sequence of suggestions on how errors for unknown types and callable in the given diagnostics can be fixed,
+        /// given the file for which those diagnostics were generated and the corresponding compilation.
+        /// The given line number is used to determine the containing namespace.
+        /// Returns an empty enumerable if any of the given arguments is null.
+        /// </summary>
+        internal static IEnumerable<(string, WorkspaceEdit)> UnknownIdSuggestions(
```

Comment:
I know these methods are internal, but it would still be good to try and follow the same naming pattern for all methods: all other methods are named SuggestionsFor* opposed to *Suggestions. Could we stick with one and have all methods follow the same pattern? To me the former is more clear in some cases. It also looks like DocCommentSuggestions is the odd one out that would be good to adapt as well.

Reply:
What do you think about the private methods that follow the *Suggestions convention, including `OpenDirectiveSuggestions` which was here before?

Reply:
All in all I think it is a good effort to have consistent naming wherever possible. I am ok with both versions, but I think we should make them all the same.

Reply:
I agree that consistency is important - the reason why I started using *Suggestions for some of them is because I think I would prefer *Suggestions over SuggestionsFor* in all cases.

Reply:
I renamed all of them to have *Suggestions at the end.